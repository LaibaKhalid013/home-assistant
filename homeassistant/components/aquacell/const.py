"""Constants for the Aquacell integration."""

DOMAIN = "aquacell"
DATA_AQUACELL = "DATA_AQUACELL"
