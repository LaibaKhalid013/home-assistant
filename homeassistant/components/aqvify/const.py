"""Constants for the Aqvify integration."""

import logging

DOMAIN = "aqvify"

CONF_DEVICE_KEY = "device_key"

LOGGER = logging.getLogger(__package__)
