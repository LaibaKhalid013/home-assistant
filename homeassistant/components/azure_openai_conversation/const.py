"""Constants for the OpenAI Conversation integration."""

import logging

DOMAIN = "azure_openai_conversation"
LOGGER = logging.getLogger(__package__)

CONF_AZURE_OPENAI_RESOURCE = "azure_openai_resource"
CONF_RECOMMENDED = "recommended"
CONF_PROMPT = "prompt"
CONF_CHAT_MODEL = "chat_model"
RECOMMENDED_CHAT_MODEL = "gpt-4"
CONF_MAX_TOKENS = "max_tokens"
RECOMMENDED_MAX_TOKENS = 150
CONF_TOP_P = "top_p"
RECOMMENDED_TOP_P = 1.0
CONF_TEMPERATURE = "temperature"
RECOMMENDED_TEMPERATURE = 1.0
AZURE_OPEN_API_VERSION = "2024-02-01"
