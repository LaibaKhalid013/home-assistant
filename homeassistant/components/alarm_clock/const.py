"""Constants for the Alarm Clock integration."""

import voluptuous as vol

from homeassistant.const import CONF_ICON, CONF_NAME
import homeassistant.helpers.config_validation as cv

DOMAIN = "alarm_clock"
ENTITY_ID_FORMAT = DOMAIN + ".{}"

ATTR_ALARM_TIME = "alarm_time"
ATTR_REPEAT_DAYS = "repeat_days"
ATTR_NEXT_ALARM = "next_alarm"

CONF_ALARM_TIME = "alarm_time"
CONF_REPEAT_DAYS = "repeat_days"

EVENT_ALARM_CLOCK_STARTED = "alarm_clock.started"
EVENT_ALARM_CLOCK_FINISHED = "alarm_clock.finished"
EVENT_ALARM_CLOCK_CANCELLED = "alarm_clock.cancelled"
EVENT_ALARM_CLOCK_CHANGED = "alarm_clock.changed"

STORAGE_KEY = DOMAIN
STORAGE_VERSION = 1

STORAGE_FIELDS = {
    vol.Required(CONF_NAME): cv.string,
    vol.Optional(CONF_ICON): cv.icon,
    vol.Required(CONF_ALARM_TIME): cv.time,
    vol.Required(CONF_REPEAT_DAYS): cv.weekdays,
}
