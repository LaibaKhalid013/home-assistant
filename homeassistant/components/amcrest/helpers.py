"""Helpers for amcrest component."""

from __future__ import annotations

import logging
from typing import TYPE_CHECKING

from .const import DOMAIN

if TYPE_CHECKING:
    from homeassistant.helpers.typing import UndefinedType


def service_signal(service: str, *args: str) -> str:
    """Encode signal."""
    return "_".join([DOMAIN, service, *args])


def log_update_error(
    logger: logging.Logger,
    action: str,
    name: str | UndefinedType | None,
    entity_type: str,
    error: Exception,
    level: int = logging.ERROR,
) -> None:
    """Log an update error."""
    logger.log(
        level,
        "Could not %s %s %s due to error: %s",
        action,
        name,
        entity_type,
        error.__class__.__name__,
    )
