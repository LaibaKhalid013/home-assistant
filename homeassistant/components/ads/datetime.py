"""Support for ADS DateTime."""

from __future__ import annotations

from datetime import UTC, datetime, timedelta

import pyads
import voluptuous as vol

from homeassistant.components.datetime import (
    PLATFORM_SCHEMA as DATETIME_PLATFORM_SCHEMA,
    DateTimeEntity,
)
from homeassistant.const import CONF_NAME
from homeassistant.core import HomeAssistant
import homeassistant.helpers.config_validation as cv
from homeassistant.helpers.entity_platform import AddEntitiesCallback
from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType

from .. import ads
from . import AdsEntity

SCAN_INTERVAL = timedelta(seconds=3)
DEFAULT_NAME = "ADS DateTime"

CONF_ADS_VAR = "adsvar"

PLATFORM_SCHEMA = DATETIME_PLATFORM_SCHEMA.extend(
    {
        vol.Optional(CONF_ADS_VAR): cv.string,
        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
    }
)


def setup_platform(
    hass: HomeAssistant,
    config: ConfigType,
    add_entities: AddEntitiesCallback,
    discovery_info: DiscoveryInfoType | None = None,
) -> None:
    """Set up the DateTime platform for ADS."""

    ads_hub = hass.data.get(ads.DATA_ADS)
    ads_var = config.get(CONF_ADS_VAR)
    name = config[CONF_NAME]

    add_entities(
        [
            AdsDateTime(
                ads_hub,
                ads_var,
                name,
            )
        ]
    )


class AdsDateTime(AdsEntity, DateTimeEntity):
    """Representation of ADS DateTime entity."""

    def __init__(self, ads_hub, ads_var, name):
        """Initialize AdsDateTime entity."""
        super().__init__(ads_hub, name, ads_var)
        self._datetime = None

    async def async_added_to_hass(self) -> None:
        """Register device notification."""

        if self._ads_var is not None:
            await self.async_initialize_device(self._ads_var, pyads.PLCTYPE_DT)
        self.async_schedule_update_ha_state(True)

    @property
    def should_poll(self) -> bool:
        """Return True if the entity should be polled."""
        return True

    async def async_set_value(self, value: datetime) -> None:
        """Set the date/time."""
        new_datetime = value
        await self.hass.async_add_executor_job(self._write_value, new_datetime)
        self._datetime = new_datetime
        self._attr_native_value = new_datetime
        self.async_write_ha_state()

    def _write_value(self, dt_value: datetime) -> None:
        # Convert Python datetime to epoch time (seconds since 1970-01-01)
        epoch_time = int(dt_value.timestamp())
        # Write the epoch time to the PLC as an integer
        self._ads_hub.write_by_name(self._ads_var, epoch_time, pyads.PLCTYPE_DT)

    async def async_update(self) -> None:
        """Retrieve the latest state from the ADS device."""
        epoch_value = self._ads_hub.read_by_name(self._ads_var, pyads.PLCTYPE_DT)
        self._datetime = datetime.fromtimestamp(epoch_value, tz=UTC)
        self._attr_native_value = self._datetime
        self.async_write_ha_state()
