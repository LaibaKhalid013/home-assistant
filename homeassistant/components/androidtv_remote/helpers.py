"""Helper functions for Android TV Remote integration."""

from __future__ import annotations

from typing import TYPE_CHECKING

from androidtvremote2 import AndroidTVRemote

from homeassistant.helpers.storage import STORAGE_DIR

from .const import CONF_ENABLE_IME, CONF_ENABLE_IME_DEFAULT_VALUE

if TYPE_CHECKING:
    from homeassistant.config_entries import ConfigEntry
    from homeassistant.core import HomeAssistant


def create_api(hass: HomeAssistant, host: str, enable_ime: bool) -> AndroidTVRemote:
    """Create an AndroidTVRemote instance."""
    return AndroidTVRemote(
        client_name="Home Assistant",
        certfile=hass.config.path(STORAGE_DIR, "androidtv_remote_cert.pem"),
        keyfile=hass.config.path(STORAGE_DIR, "androidtv_remote_key.pem"),
        host=host,
        loop=hass.loop,
        enable_ime=enable_ime,
    )


def get_enable_ime(entry: ConfigEntry) -> bool:
    """Get value of enable_ime option or its default value."""
    return entry.options.get(CONF_ENABLE_IME, CONF_ENABLE_IME_DEFAULT_VALUE)
