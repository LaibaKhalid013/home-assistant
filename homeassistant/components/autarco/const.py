"""Constants for the Autarco integration."""

from __future__ import annotations

from datetime import timedelta
import logging
from typing import Final

DOMAIN: Final = "autarco"
LOGGER = logging.getLogger(__package__)
SCAN_INTERVAL = timedelta(minutes=1)

CONF_PUBLIC_KEY: Final = "public_key"

SENSORS_SOLAR: Final = "solar"
