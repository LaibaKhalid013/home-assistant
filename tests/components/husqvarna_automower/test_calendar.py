"""Tests for calendar platform."""

from datetime import UTC, datetime
from unittest.mock import AsyncMock

import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.calendar import (
    DOMAIN as CALENDAR_DOMAIN,
    EVENT_END_DATETIME,
    EVENT_START_DATETIME,
    SERVICE_GET_EVENTS,
)
from homeassistant.const import ATTR_ENTITY_ID
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import setup_integration

from tests.common import MockConfigEntry


@pytest.mark.freeze_time(datetime(2023, 6, 5, 12, tzinfo=UTC))
@pytest.mark.parametrize(
    (
        "start_date",
        "end_date",
    ),
    [
        (datetime(2024, 3, 2, 6, 0), datetime(2024, 3, 18, 6, 0)),
    ],
)
async def test_calendar_snapshot(
    hass: HomeAssistant,
    entity_registry: er.EntityRegistry,
    mock_automower_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
    start_date: datetime,
    end_date: datetime,
) -> None:
    """Snapshot tesst of the calendars."""
    await hass.config.async_set_time_zone("Europe/Berlin")
    await setup_integration(hass, mock_config_entry)
    events = await hass.services.async_call(
        CALENDAR_DOMAIN,
        SERVICE_GET_EVENTS,
        {
            ATTR_ENTITY_ID: "calendar.test_mower_1",
            EVENT_START_DATETIME: start_date,
            EVENT_END_DATETIME: end_date,
        },
        blocking=True,
        return_response=True,
    )

    assert events == snapshot
