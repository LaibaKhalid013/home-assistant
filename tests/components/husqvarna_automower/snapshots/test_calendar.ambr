# serializer version: 1
# name: test_binary_sensor_snapshot[start_date0-end_date0]
  dict({
    'calendar.test_mower_1': dict({
      'events': list([
        dict({
          'end': '2024-03-04T16:00:00-08:00',
          'start': '2024-03-04T11:00:00-08:00',
          'summary': 'Front lawn Schedule 1',
        }),
        dict({
          'end': '2024-03-05T00:00:00-08:00',
          'start': '2024-03-04T16:00:00-08:00',
          'summary': 'Front lawn Schedule 2',
        }),
        dict({
          'end': '2024-03-06T16:00:00-08:00',
          'start': '2024-03-06T11:00:00-08:00',
          'summary': 'Front lawn Schedule 1',
        }),
        dict({
          'end': '2024-03-07T00:00:00-08:00',
          'start': '2024-03-06T16:00:00-08:00',
          'summary': 'Front lawn Schedule 2',
        }),
        dict({
          'end': '2024-03-08T16:00:00-08:00',
          'start': '2024-03-08T11:00:00-08:00',
          'summary': 'Front lawn Schedule 1',
        }),
        dict({
          'end': '2024-03-09T00:00:00-08:00',
          'start': '2024-03-08T16:00:00-08:00',
          'summary': 'Front lawn Schedule 2',
        }),
        dict({
          'end': '2024-03-11T17:00:00-07:00',
          'start': '2024-03-11T12:00:00-07:00',
          'summary': 'Front lawn Schedule 1',
        }),
        dict({
          'end': '2024-03-12T01:00:00-07:00',
          'start': '2024-03-11T17:00:00-07:00',
          'summary': 'Front lawn Schedule 2',
        }),
        dict({
          'end': '2024-03-13T17:00:00-07:00',
          'start': '2024-03-13T12:00:00-07:00',
          'summary': 'Front lawn Schedule 1',
        }),
        dict({
          'end': '2024-03-14T01:00:00-07:00',
          'start': '2024-03-13T17:00:00-07:00',
          'summary': 'Front lawn Schedule 2',
        }),
        dict({
          'end': '2024-03-15T17:00:00-07:00',
          'start': '2024-03-15T12:00:00-07:00',
          'summary': 'Front lawn Schedule 1',
        }),
        dict({
          'end': '2024-03-16T01:00:00-07:00',
          'start': '2024-03-15T17:00:00-07:00',
          'summary': 'Front lawn Schedule 2',
        }),
      ]),
    }),
  })
# ---
